sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2154,6 +2155,18 @@ class Dataset[T] private[sql](
       colNames,
       "in given column names",
       sparkSession.sessionState.conf.caseSensitiveAnalysis)
+    var numProjects = 0
+    var currPlan = logicalPlan
+    while (currPlan.isInstanceOf[Project] && numProjects < 50) {
```

Comment:
Yep. If we need to warn, +1 for adding new configuration instead of `50` here and line 2164.

`50` looks effective to detect this pattern, but can we have a higher value which is more practically related to the warning messages(performance degradation or OOM?)?

Reply:
yes, I just wanted to be sure that we agree on the idea. Do you have hint/preferences for the name of the config?
I didn't want to introduce a high value in order not to have a high impact on perf for the loop to check this. What do you think?

Reply:
How about checking the count of continuous projection and keep/reduce the count? I can't imagine end users to create more than (like) 20 times projection without withColumn/drop/etc instead of select.

Reply:
What do you mean @HeartSaVioR ? I don't think it is a good idea to add a counter in the Dataset class, which, moreover, should be carried over when creating a new Dataset, otherwise it is useless. It'd be an overkill for this IMO.

Reply:
My bad. Just re-read the code (while loop) and now seeing that this implementation already considers only continuous projections. Sorry about confusion.

Reply:
np, thanks for your comment